velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -259,46 +314,80 @@ struct DateTruncFunction {
     }
   }
 
-  FOLLY_ALWAYS_INLINE bool call(
-      out_type<Timestamp>& result,
-      const arg_type<Varchar>& unitString,
-      const arg_type<Timestamp>& timestamp) {
-    const auto unit = unit_.has_value()
-        ? unit_.value()
-        : fromDateTimeUnitString(unitString, true /*throwIfInvalid*/).value();
-    if (unit == DateTimeUnit::kSecond) {
-      result = Timestamp(timestamp.getSeconds(), 0);
-      return true;
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* unitString,
+      const arg_type<Date>* /*date*/) {
+    if (unitString != nullptr) {
+      unit_ = fromDateTimeUnitString(*unitString, false /*throwIfInvalid*/);
     }
+  }
 
-    auto dateTime = getDateTime(timestamp, timeZone_);
+  FOLLY_ALWAYS_INLINE void fixTmForTruncation(
```

Comment:
nit: do not abbreviate; tmValue -> dateTime and fixTmForTruncation -> adjustDateTime (no need to repeat "truncation" since this method is already in the context of date truncation)